dataplane/standalone/apigen/apigen.go

```diff
@@ -260,6 +261,248 @@ var (
 	funcExpr = regexp.MustCompile(`^([a-z]*_)(\w*)_(attribute|stats_ext|stats)|([a-z]*)_(\w*)$`)
 )
 
+var protoTmpl = template.Must(template.New("cc").Parse(`
+syntax = "proto3";
+
+package lemming.dataplane.sai;
+
+option go_package = "github.com/openconfig/lemming/proto/dataplane/sai";
+
+{{ range .Messages }}
+message {{ .RequestName }} {
+	{{ .RequestFieldsWrapperStart -}}
+	{{- range .RequestFields }}
+	{{ .ProtoType }} {{ .Name }} = {{ .Index }};
+	{{- end }}
+	{{ .RequestFieldsWrapperEnd }}
+}
+
+message {{ .ResponseName }} {
+	{{ .ResponseFieldsWrapperStart -}}
+	{{- range .ResponseFields }}
+	{{ .ProtoType }} {{ .Name }} = {{ .Index }};
+	{{- end }}
+	{{ .ResponseFieldsWrapperEnd }}
+}
+
+{{ end }}
+
+
+service {{ .ServiceName }} {
+	{{- range .RPCs }}
+	rpc {{ .Name }} ({{ .RequestName }}) returns ({{ .ResponseName }}) {}
+	{{- end }}
+}
+`))
+
+type protoTmplData struct {
+	Messages    []protoTmplMessage
+	RPCs        []protoRPC
+	ServiceName string
+}
+
+type protoTmplMessage struct {
+	RequestName                string
+	ResponseName               string
+	RequestFieldsWrapperStart  string
+	RequestFieldsWrapperEnd    string
+	RequestFields              []protoTmplField
+	ResponseFieldsWrapperStart string
+	ResponseFieldsWrapperEnd   string
+	ResponseFields             []protoTmplField
+}
+
+type protoTmplField struct {
+	ProtoType string
+	Name      string
+	Index     int
+}
+
+type protoRPC struct {
+	RequestName  string
+	ResponseName string
+	Name         string
+}
+
+type saiTypeInfo struct {
+	Repeated  bool
+	Primitive bool
+	ProtoType string
+}
+
+var saiTypeToProto = map[string]saiTypeInfo{
+	"bool": {
+		ProtoType: "bool",
```

Comment:
Is this not a primitive?

Reply:
removed primitive field, not sure i need it yet